src/persist/src/operators/mod.rs

```diff
@@ -24,3 +39,121 @@ pub(crate) fn split_ok_err<K, V>(
         (Err(err), ts, diff) => Err((err, ts, diff)),
     }
 }
+
+/// A helper type to integrate yielding to timely with async futures. Its intended to be used with
+/// a top level while loop. Timely will automatically make the returned futures resolve when there
+/// are progress or data updates to be processed by the operator
+///
+/// ```ignore
+/// async fn example(s: Scheduler) {
+///     while scheduler.yield_now().await {
+///     }
+/// }
+/// ```
+#[derive(Debug)]
+pub struct Scheduler(Rc<Cell<bool>>);
+
+/// The future returned by [Scheduler::yield_now]
+#[derive(Debug)]
+pub struct Yield<'a>(&'a Rc<Cell<bool>>);
+
+impl Scheduler {
+    /// Returns a future that will stay pending until timely reschedules the operator
+    pub fn yield_now(&self) -> Yield {
+        Yield(&self.0)
+    }
+}
+
+impl Future for Yield<'_> {
+    type Output = bool;
+    fn poll(self: Pin<&mut Self>, _cx: &mut Context) -> Poll<Self::Output> {
+        if self.0.replace(false) {
+            Poll::Ready(true)
+        } else {
+            Poll::Pending
+        }
+    }
+}
+
+trait OpeatorBuilderExt<G: Scope> {
```

Comment:
nit: Operator

Reply:
oops, I don't know how I manage to mispell it three times :P fixed